js/form_editor_controller.js

```diff
@@ -451,8 +451,8 @@ class GlpiFormEditorController
 
             // Update input name
             let postfix = "";
-            if (typeof field === 'string' && field.match(/\[([\w[\]]*)\]$/g)) {
-                postfix = field.match(/\[([\w[\]]*)\]$/g);
+            if (typeof field === 'string' && field.match(/\[([\w-[\]]*)\]$/g)) {
+                postfix = field.match(/\[([\w-[\]]*)\]$/g);
```

Comment:
1. Pattern could be placed in a constant.
2. The `RegExp.test()` method could be used in the `if` condition.
```suggestion
            cons postfix_pattern = new RegExp('\[([\w-\[\]]*)\]$', 'g');
            if (typeof field === 'string' && postfix_pattern.test(field)) {
                postfix = field.match(postfix_pattern);
```